src/west/cmd/project.py

```diff
@@ -513,6 +520,9 @@ def _fetch(project):
         _inf(project, msg)
         _git_base(project, cmd)
 
+        if checkout:
+            _checkout(project, project.revision)
```

Comment:
The checking out was more a side effect of using `git clone`. Not sure what the initial repo state should be later, with `git init` + `git remote add` + `git fetch` (which seems to be what repo does too). Could either leave nothing checked out, or check out e.g. `manifest-rev` (or a detached HEAD at `manifest-rev`).

Doesn't hurt in the meantime though.

Reply:
> (or a detached HEAD at manifest-rev)

That's the behavior this patch enforces.

Reply:
Yeah, though it's inconsistent now. This code will change soon anyway though, so it doesn't matter.

Reply:
Later on you will consistently get either a detached HEAD, or nothing checked out at all (the `git init` state... think HEAD just points at a non-existing `refs/heads/master` branch there, which gets created when you make the first commit).

At the moment, you'll either get a detached HEAD or a tracking branch. Initially, that is. Once you create a branch, it should look the same.